packages/lesswrong/components/search/ExpandedCommentsSearchHit.tsx

```diff
@@ -0,0 +1,127 @@
+// eslint-disable-next-line no-restricted-imports
+import { useHistory } from 'react-router-dom';
+
+const styles = (theme: ThemeType): JssStyles => ({
+  root: {
+    maxWidth: 600,
+    paddingTop: 2,
+    paddingBottom: 2,
+    marginBottom: 18,
+    cursor: 'pointer',
+    '&:hover': {
+      opacity: 0.5
+    }
+  },
+  link: {
+    '&:hover': {
+      opacity: 1
+    }
+  },
+  authorRow: {
+    display: "flex",
+    flexWrap: 'wrap',
+    alignItems: 'baseline',
+    columnGap: 16,
+    rowGap: '3px',
+    color: theme.palette.grey[600],
+    fontSize: 12,
+    fontFamily: theme.typography.fontFamily,
+    marginTop: 6
+  },
+  metaInfo: {
+    display: "flex",
+    alignItems: 'center',
+    columnGap: 3
+  },
+  title: {
+    display: 'flex',
+    alignItems: 'center',
+    columnGap: 6,
+    fontSize: 15,
+    lineHeight: '22px',
+    fontFamily: theme.typography.postStyle.fontFamily,
+    color: theme.palette.grey[800],
+    fontWeight: 600,
+  },
+  tagIcon: {
+    fontSize: 14,
+    color: theme.palette.grey[600],
+  },
+  snippet: {
+    overflowWrap: "break-word",
+    fontFamily: theme.typography.fontFamily,
+    wordBreak: "break-word",
+    fontSize: 14,
+    lineHeight: '21px',
+    color: theme.palette.grey[700],
+    marginTop: 5
+  }
+})
+
+const ExpandedCommentsSearchHit = ({hit, classes}: {
+  hit: Hit<any>,
+  classes: ClassesType,
+}) => {
+  const history = useHistory()
+
+  const { FormatDate, UserNameDeleted } = Components
+  const comment: AlgoliaComment = hit
+  
+  let url = "";
+  if (comment.postId && comment.postSlug) {
+    url = `${postGetPageUrl({
+      _id: comment.postId ?? "",
+      slug: comment.postSlug ?? "",
+      isEvent: comment.postIsEvent,
+      groupId: comment.postGroupId,
+    })}#${comment._id}`
+  } else if (comment.tagSlug && comment.tagCommentType) {
+    url = tagGetCommentLink(comment.tagSlug, comment._id, comment.tagCommentType as TagCommentType)
+  }
+  
+  const handleClick = () => {
+    history.push(url)
+  }
+
+  return <div className={classes.root} onClick={handleClick}>
+    <Link to={url} className={classes.link} onClick={(e) => e.stopPropagation()}>
+      {comment.postTitle && <div className={classes.title}>
+        {comment.postTitle}
+      </div>}
+      {!comment.postTitle && comment.tagName && <div className={classes.title}>
+        <TagIcon className={classes.tagIcon} />
+        {comment.tagName}
+      </div>}
+      <div className={classes.snippet}>
+        <Snippet className={classes.snippet} attribute="body" hit={comment} tagName="mark" />
+      </div>
+    </Link>
+    <div className={classes.authorRow}>
+      {comment.authorSlug ? <Link to={userGetProfileUrlFromSlug(comment.authorSlug)} onClick={(e) => e.stopPropagation()}>
+        {comment.authorDisplayName}
+      </Link> : <UserNameDeleted />}
+      <span>{comment.baseScore ?? 0} karma</span>
+      <FormatDate date={comment.createdAt} />
+    </div>
+  </div>
+}
```

Comment:
Could you explain why these `e. stopPropagation()`s are needed

Reply:
Yeah so this was annoying - basically I wanted to make the whole search result clickable to go to the comment, but also clicking on the author name should go to their profile. You can't have nested anchor tags, so I faked the "whole result is clickable" part. But I want to make sure that clicking on an actual link doesn't trigger the outer action, hence the `stopPropagation`.